overlord/snapstate/snapmgr.go

```diff
@@ -21,38 +21,80 @@
 package snapstate
 
 import (
+	"fmt"
+
 	"github.com/ubuntu-core/snappy/overlord/state"
+	"github.com/ubuntu-core/snappy/progress"
+	"github.com/ubuntu-core/snappy/snappy"
 )
 
+// Install initiates a change installing snap.
+func Install(change *state.Change, snap string) error {
+	change.State().Lock()
+	defer change.State().Unlock()
+
+	tIns := change.NewTask("install-snap", fmt.Sprintf("Installing %q", snap))
```

Comment:
Gustavo recommended to apply i18n to tasks as those are going to be user visible.